xline-client/tests/auth.rs

```diff
@@ -0,0 +1,178 @@
+//! The following tests are originally from `etcd-client`
+use common::get_cluster_client;
+use xline_client::{
+    error::Result,
+    types::auth::{
+        AuthRoleAddRequest, AuthRoleDeleteRequest, AuthRoleGetRequest,
+        AuthRoleGrantPermissionRequest, AuthRoleRevokePermissionRequest, AuthUserAddRequest,
+        AuthUserChangePasswordRequest, AuthUserDeleteRequest, AuthUserGetRequest,
+        AuthUserGrantRoleRequest, AuthUserRevokeRoleRequest, Permission,
+    },
+};
+use xlineapi::Type;
+
+mod common;
+
+#[tokio::test]
+async fn role() -> Result<()> {
```

Comment:
A big test is not a good practice. Please divide it into serval small and concrete tests with meaningful names.

Reply:
done